tree/ntuple/inc/ROOT/RPageStorage.hxx

```diff
@@ -313,6 +313,8 @@ public:
 
    virtual ROOT::NTupleSize_t GetNEntries() const = 0;
 
+   virtual TDirectory *GetUnderlyingDirectory() const { return nullptr; }
```

Comment:
This breaks the abstraction of `RPageSink` quite heavily, since not all sinks are a file and even of those not all have a `TDirectory`. I'm not sure this makes sense to ask a `TDirectory` from all page sinks...